provider/provider_program_test.go

```diff
@@ -0,0 +1,119 @@
+//go:build !go && !nodejs && !python && !dotnet
+// +build !go,!nodejs,!python,!dotnet
+
+package github
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/pulumi/providertest"
+	"github.com/pulumi/providertest/optproviderupgrade"
+	"github.com/pulumi/providertest/pulumitest"
+	"github.com/pulumi/providertest/pulumitest/assertpreview"
+	"github.com/pulumi/providertest/pulumitest/opttest"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+)
+
+const providerName = "github"
+const defaultBaselineVersion = "6.2.0"
+
+var programs = []string{
+	"test-programs/index_repositoryautolinkreference",
+	"test-programs/index_repositorywebhook",
+}
+
+func TestUpgradeCoverage(t *testing.T) {
+	providertest.ReportUpgradeCoverage(t)
+}
+
+type UpgradeTestOpts struct {
+	baselineVersion string
+	assertFunc      func(*testing.T, auto.PreviewResult)
+	config          map[string]string
+}
+
+func WithBaselineVersion(baselineVersion string) func(opts *UpgradeTestOpts) {
+	return func(opts *UpgradeTestOpts) {
+		opts.baselineVersion = baselineVersion
+	}
+}
+
+func WithAssertFunc(assertFunc func(*testing.T, auto.PreviewResult)) func(opts *UpgradeTestOpts) {
+	return func(opts *UpgradeTestOpts) {
+		opts.assertFunc = assertFunc
+	}
+}
+
+func WithConfig(config map[string]string) func(opts *UpgradeTestOpts) {
+	return func(opts *UpgradeTestOpts) {
+		opts.config = config
+	}
+}
+func testProviderUpgrade(t *testing.T, dir string, opts ...func(*UpgradeTestOpts)) {
+	options := &UpgradeTestOpts{}
+	for _, o := range opts {
+		o(options)
+	}
+	testProviderUpgradeWithOpts(t, dir, options.baselineVersion, options.config, options.assertFunc)
+}
+
+func testProviderUpgradeWithOpts(
+	t *testing.T, dir, baselineVersion string, config map[string]string,
+	assertFunction func(*testing.T, auto.PreviewResult),
+) {
+	if testing.Short() {
+		t.Skipf("Skipping in testing.Short() mode, assuming this is a CI run without credentials")
+	}
+	cwd, err := os.Getwd()
+	require.NoError(t, err)
+	if baselineVersion == "" {
+		baselineVersion = defaultBaselineVersion
+	}
+	test := pulumitest.NewPulumiTest(t, dir,
+		opttest.DownloadProviderVersion(providerName, baselineVersion),
+		opttest.LocalProviderPath(providerName, filepath.Join(cwd, "..", "bin")),
```

Comment:
You'll want to set `github:token` and `github:owner` to `GITHUB_TESTING_TOKEN` and `GITHUB_ORG`, respectively. Then CI should pass.

See how integration tests are set up: 
https://github.com/pulumi/pulumi-github/blob/master/examples/examples_test.go#L15